### PR TITLE
Adds static analysis/code style checker Flake8

### PR DESCRIPTION
**Pull request by happz**
With the default setup, as part of pre-commit checks. Trivial changes
reported and fixed in the patch, mostly unused imports/variables, lines
too long and unnecessary f-strings.

### CODE REVIEW

**Comment by happz** [comment]
I added a note to TMT hacking session notes, a proposal to add this tool, but then I asked myself how hard would it be to sort the reported issues out, and it proved to be surprisingly easy to get rid of almost all of them. So, let's see what's going to happen next.

Pre-commit is going to fail because there is a genuine issue in an example plugin.

**Comment by lukaszachy** [comment]
@happz Thank you for flake8, I've pushed commit fixing example - pre-commit run --all is now green.

**Comment by lukaszachy** [comment]
Hm. unit tests are failing for me:
```
FAILED tests/unit/test_cli.py::test_create - assert 2 == 0
FAILED tests/unit/test_cli.py::test_systemd - assert 2 == 0
```

@happz Can you reproduce it?
Looks to be connected with cli command renamed. (`"Usage: main test [OPTIONS] COMMAND [ARGS]...\nTry 'main test --help' for help.\n\nError: No such command 'create'.\n"`

**Comment by lukaszachy** [comment]
Right. Simplest reproducer is `tmt plans show` - Command has changed to plans-show.

```
tmt plans --help
Usage: tmt plans [OPTIONS] COMMAND [ARGS]...

  Manage test plans (L2 metadata).

  Search for available plans.
  Explore detailed test step configuration.

Options:
  -v, --verbose  Show more details. Use multiple times to raise verbosity.
  -d, --debug    Provide debugging information. Repeat to see more details.
  -q, --quiet    Be quiet. Exit code is just enough for me.
  --help         Show this message and exit.

Commands:
  plans-ls      List available plans.
  plans-show    Show plan details.
  lint          Check plans against the L2 metadata specification.
  plans-create  Create a new plan based on given template.
  plans-export  Export plans into desired format.
  id            Generate a unique id for each selected plan.

```

**Comment by lukaszachy** [comment]
@happz should be fixed now, lets see if test/full shows something.

**Review by lukaszachy — APPROVED** [review]
LGTM however let's wait on @psss approval.

File: tests/unit/test_report_junit.py
Comment (happz) [reply tests/unit/test_report_junit.py:71]: It complains when running on my laptop, because of line 67 which is over 90 characters long:

```
 pytest -m flake8 --flake8 tests/unit/test_report_junit.py 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/happz/git/tmt/tests/unit, configfile: pytest.ini
plugins: flake8-1.1.1
collected 6 items / 5 deselected / 1 selected                                                                                                                                                

tests/unit/test_report_junit.py F                                                                                                                                                      [100%]

========================================================================================== FAILURES ==========================================================================================
________________________________________________________________________________________ FLAKE8-check ________________________________________________________________________________________
/home/happz/git/tmt/tests/unit/test_report_junit.py:1:1: F401 'unittest.mock.patch' imported but unused
/home/happz/git/tmt/tests/unit/test_report_junit.py:67:80: E501 line too long (95 > 79 characters)
/home/happz/git/tmt/tests/unit/test_report_junit.py:86:80: E501 line too long (95 > 79 characters)
/home/happz/git/tmt/tests/unit/test_report_junit.py:107:80: E501 line too long (95 > 79 characters)
/home/happz/git/tmt/tests/unit/test_report_junit.py:128:80: E501 line too long (95 > 79 characters)
/home/happz/git/tmt/tests/unit/test_report_junit.py:149:80: E501 line too long (95 > 79 characters)
```

Comment (psss) [reply tests/unit/test_report_junit.py:71]: Ah, sorry, did not realize I already included a `.flake8` config with the `99` chars max. Will add it to the PR.

Comment (happz) [reply tests/unit/test_report_junit.py:71]: Ok, if you want to incorporate such a change into this patch, there are a couple more `E501` to remove then.

Comment (psss) [reply tests/unit/test_report_junit.py:71]: Yes, I'm already working on that.

**Review by psss — APPROVED** [review]
Thanks for preparing this. Looks good. I just simplified the discover plugin example a bit (@lukaszachy, is that ok?), made the method naming in `cli.py` a bit more consistent and also included docs and config for the newly agreed `99` chars limit.

**Comment by lukaszachy** [comment]
@psss Sure, as long as discover example works to some extent.

**Pull request merged**
